Model/Risk/Addresscheck.php

```diff
@@ -0,0 +1,275 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with PAYONE Magento 2 Connector. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * PHP version 5
+ *
+ * @category  Payone
+ * @package   Payone_Magento2_Plugin
+ * @author    FATCHIP GmbH <support@fatchip.de>
+ * @copyright 2003 - 2016 Payone GmbH
+ * @license   <http://www.gnu.org/licenses/> GNU Lesser General Public License
+ * @link      http://www.payone.de
+ */
+
+namespace Payone\Core\Model\Risk;
+
+use Magento\Quote\Api\Data\AddressInterface;
+
+/**
+ * Model for calling the addresscheck request
+ */
+class Addresscheck
+{
+    /**
+     * Response from the PAYONE addresscheck request
+     *
+     * @var array
+     */
+    protected $aResponse = null;
+
+    /**
+     * Determines if the request was NOT executed because the lifetime of the last was still valid
+     *
+     * @var bool
+     */
+    protected $blIsLifetimeValid;
+
+    /**
+     * Saves if the address was corrected
+     *
+     * @var bool
+     */
+    protected $addressCorrected;
+
+    /**
+     * PAYONE addresscheck request model
+     *
+     * @var \Payone\Core\Model\Api\Request\Addresscheck
+     */
+    protected $addresscheck;
+
+    /**
+     * PAYONE database helper
+     *
+     * @var \Payone\Core\Helper\Database
+     */
+    protected $databaseHelper;
+
+    /**
+     * PAYONE toolkit helper
+     *
+     * @var \Payone\Core\Helper\Toolkit
+     */
+    protected $toolkitHelper;
+
+    /**
+     * Constructor
+     *
+     * @param \Payone\Core\Model\Api\Request\Addresscheck $addresscheck
+     * @param \Payone\Core\Helper\Database                $databaseHelper
+     * @param \Payone\Core\Helper\Toolkit                 $toolkitHelper
+     */
+    public function __construct(
+        \Payone\Core\Model\Api\Request\Addresscheck $addresscheck,
+        \Payone\Core\Helper\Database $databaseHelper,
+        \Payone\Core\Helper\Toolkit $toolkitHelper
+    ) {
+        $this->addresscheck = $addresscheck;
+        $this->databaseHelper = $databaseHelper;
+        $this->toolkitHelper = $toolkitHelper;
+    }
+
+    /**
+     * Return if check was not executed because the lifetime of the last check was still valid
+     *
+     * @return bool
+     */
+    public function isLifetimeStillValid()
+    {
+        return $this->blIsLifetimeValid;
+    }
+
+    /**
+     * Return addressCorrected property
+     *
+     * @return bool
+     */
+    public function isAddressCorrected()
+    {
+        return $this->addressCorrected;
+    }
+
+    /**
+     * Change the address according to the response
+     *
+     * @param  AddressInterface $oAddress
+     * @return AddressInterface
+     */
+    public function correctAddress(AddressInterface $oAddress)
```

Comment:
This method is too complex, please refactor: https://scrutinizer-ci.com/g/PAYONE-GmbH/magento-2/inspections/f57e816f-b8e5-4e29-b0a2-aa3d5399368d/code-structure/operation/Payone%5CCore%5CModel%5CRisk%5CAddresscheck%3A%3AcorrectAddress